### PR TITLE
Added References to jQuery Validate Plugins in the Model Validation Topic

### LINKED ISSUE #4750: Reference jQuery Validate Documentation
The jQuery Validate documentation on the [remote method](https://jqueryvalidation.org/remote-method/) gives very specific details on what values the serverside resource should return.  The ASP.NET Core docs currently should mention those details or at least provide a link to the jQuery Validate documentation.

## Suggested Reviewers:
@rachelappel 
@Rick-Anderson 
@isaac2004 
@guardrex 
@v-anpasi

### PR DESCRIPTION
**Pull request by Rabadash8820**
I felt like the model validation topic needed to explain better the link between validation attributes, tag helpers, [jQuery unobtrusive validation](https://github.com/aspnet/jquery-validation-unobtrusive), and the [jQuery Validate](https://jqueryvalidation.org/) plugin.  These technologies interact in complex ways, and I found myself having to dig all around the internet to understand these interactions while developing my own apps.  It would've been nice to have a high-level introduction (with links) in the ASP.NET Core docs themselves.

  Fixes #4750

## Suggested Reviewers:
@rachelappel 
@Rick-Anderson 
@isaac2004 
@guardrex 
@v-anpasi

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@Rabadash8820  nice improvement. @rachelappel  can you provide a edit and shorten some of the sentences?

I think @rachelappel  is out for a few days so it might take awhile.

File: aspnetcore/mvc/models/validation.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ The tag helpers above render the HTML below. Notice that the `data-` attributes
 </form>
 ```
 
-Client-side validation prevents submission until the form is valid. The Submit button runs JavaScript that either submits the form or displays error messages.
+Thus, client-side validation prevents submission until the form is valid. The Submit button runs JavaScript that either submits the form or displays error messages.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/models/validation.md]: ~Thus,~

Comment (Rabadash8820) [reply aspnetcore/mvc/models/validation.md]: Thought it added a nice oomph, but no worries :) I'll wait for @rachelappel's feedback before making any more commits.

**Comment by Rick-Anderson** [comment]
@rachelappel  please review.

**Pull request merged**